docs/docker-swarm-traefik-registry.md

```diff
@@ -0,0 +1,553 @@
+
+Connect to your remote server as normally, e.g.:
+
+```bash
+ssh root@172.173.174.175
+```
+
+Create a backup of your SSH config file:
+
+```bash
+cp /etc/ssh/sshd_config /etc/ssh/sshd_config.backup
+```
+
+Modify your SSH config. 
+
+**Warning**: if something is broken after modifying the SSH configuration, you could lock yourself out of the server.
+
+
+You need to change the line `Port 22` to `Port 2222`. You can do it with this command:
+
+```bash
+sed -i 's|Port 22$|Port 2222|' /etc/ssh/sshd_config
+```
+
+Or you can modify it with `nano` by hand, with:
+
+```bash
+nano /etc/ssh/sshd_config
+```
+
+Then restart the SSH server:
+
+```bash
+systemctl restart sshd.service
+```
+
+**Warning**: at this point, if you lose your connection and something was wrong in the configuration, you could lock yourself out of the server. Run the following steps in a new terminal session, without closing the existing one, so that, if something was wrong, you can use the current session to edit the configurations, revert them, and restart the SSH service, before being locked out.
+
+In a different terminal session, without closing the existing one, try connecting with SSH to your server using the new port, e.g.:
+
+```bash
+ssh -p 2222 root@172.173.174.175
+```
+
+If you get connected to the remote server normally, everything is fine.
+
+
+## Set up swarm mode
+
+In Docker Swarm Mode you have one or more "manager" nodes and one or more "worker" nodes (that can be the same manager nodes).
+
+The first step, is to configure one (or more) manager nodes.
+
+* On the main manager node, run:
+
+```bash
+docker swarm init
+```
+
+* On the main manager node, for each additional manager node you want to set up, run:
+
+```bash
+docker swarm join-token manager
+```
+
+* Copy the result and paste it in the additional manager node's terminal, it will be something like:
+
+```bash
+ docker swarm join --token SWMTKN-1-5tl7yaasdfd9qt9j0easdfnml4lqbosbasf14p13-f3hem9ckmkhasdf3idrzk5gz 172.173.174.175:2377
+```
+
+* On the main manager node, for each additional worker node you want to set up, run:
+
+```bash
+docker swarm join-token worker
+```
+
+* Copy the result and paste it in the additional worker node's terminal, it will be something like:
+
+```bash
+docker swarm join --token SWMTKN-1-5tl7ya98erd9qtasdfml4lqbosbhfqv3asdf4p13-dzw6ugasdfk0arn0 172.173.174.175:2377
+```
+
+## Traefik
+
+Set up a main load balancer with Traefik that handles the public connections and Let's encrypt HTTPS certificates.
+
+* Create a network that will be shared with Traefik and the containers that should be accessible from the outside, with:
+
+```bash
+docker network create --driver=overlay traefik-public
+```
+
+* Create an environment variable with your email, to be used for the generation of Let's Encrypt certificates:
+
+```bash
+export EMAIL=admin@example.com
+```
+
+* Create a Traefik service:
+
+```bash
+docker service create \
```

Comment:
In general instead of a long command. I would also like to see a separate `docker-compose.traefik.stack.yml`. It's better to maintain instead of a command.

Reply:
Sure. I updated it to have an external extra stack file for Traefik. And I updated the docs accordingly.

I also updated those same docs with the latest changes/updates/improvements from [my original post](https://github.com/tiangolo/medium-posts/tree/master/docker-swarm-mode-and-traefik-for-a-https-cluster).